### ISSUE TITLE
Xcode 8 (b3) + Swift 2.3: Carthage build fails but normal build works.

### ISSUE THREAD

**Issue by alpennec** [Carthage/Carthage#1396]
* `carthage version`: 0.17.2
* `xcodebuild -version`: Xcode 8.0 // Build version 8S162m (but  TOOLCHAINS=com.apple.dt.toolchain.Swift_2_3 to build using Swift 2.3)
* Are you using `--no-build`? No
* Are you using `--no-use-binaries`? No
* Are you using `--use-submodules`? No

**Cartfile**
```
github "Alamofire/Alamofire" "swift2.3"
github "Alamofire/AlamofireNetworkActivityIndicator" "swift2.3"
github "duemunk/Async"
github "xmartlabs/Eureka" "swift2.3"
github "Haneke/HanekeSwift"
github "JonasGessner/JGProgressHUD"
github "mixpanel/mixpanel-iphone"
github "mac-cain13/R.swift.Library"
github "malcommac/SwiftDate"
github "SwiftyBeaver/SwiftyBeaver"
github "SwiftyJSON/SwiftyJSON"
```

I don't understand why I can build the framework with Xcode if I open the project but I can't do it using Carthage. The code is correct if I open it in Xcode. Is that related to Eureka framework? I [opened an issue](https://github.com/xmartlabs/Eureka/issues/567) but apparently it's maybe due to Carthage? 

One developer suggests 
> When compiling Source/Helpers.swift in Xcode you will see that this builds correctly.
Seems that Carthage is using another version of Swift compiler. I do not know how to change this but you must make sure that you are using the compiler from Xcode 8 beta 2.

**Carthage Output**
```
AxeleFounders3:Skipper Axel$ TOOLCHAINS=com.apple.dt.toolchain.Swift_2_3 carthage update --platform iOS

*** xcodebuild output can be found in /var/folders/01/z3cm_qz957d4jwx6f54_f6pr0000gn/T/carthage-xcodebuild.AWR8hW.log
*** Building scheme "Alamofire iOS" in Alamofire.xcworkspace
*** Building scheme "AlamofireNetworkActivityIndicator" in AlamofireNetworkActivityIndicator.xcodeproj
*** Building scheme "Pods-AsyncExample iOS-AsyncSwift" in AsyncPodsExample.xcworkspace
*** Building scheme "Async iOS" in Async.xcodeproj
*** Building scheme "Eureka" in Eureka.xcworkspace
--- xcodebuild: WARNING: Unable to open project file '/Users/Axel/Developer/Skipper/Carthage/Checkouts/Eureka/Eureka.playground' in workspace '/Users/Axel/Developer/Skipper/Carthage/Checkouts/Eureka/Eureka.xcworkspace'.
** BUILD FAILED **


The following build commands failed:
	CompileSwift normal arm64
	CompileSwiftSources normal arm64 com.apple.xcode.tools.swift.compiler
(2 failures)
A shell task (/usr/bin/xcrun xcodebuild -workspace /Users/Axel/Developer/Skipper/Carthage/Checkouts/Eureka/Eureka.xcworkspace -scheme Eureka -configuration Release -sdk iphoneos ONLY_ACTIVE_ARCH=NO BITCODE_GENERATION_MODE=bitcode CODE_SIGNING_REQUIRED=NO CODE_SIGN_IDENTITY= CARTHAGE=YES clean build) failed with exit code 65:
--- xcodebuild: WARNING: Unable to open project file '/Users/Axel/Developer/Skipper/Carthage/Checkouts/Eureka/Eureka.playground' in workspace '/Users/Axel/Developer/Skipper/Carthage/Checkouts/Eureka/Eureka.xcworkspace'.
** BUILD FAILED **


The following build commands failed:
	CompileSwift normal arm64
	CompileSwiftSources normal arm64 com.apple.xcode.tools.swift.compiler
(2 failures)
```

Thanks,
Axel

**Comment by benjohnde** [comment]
It seems that it is connected with the following issue: https://forums.developer.apple.com/thread/52139

If I jump right into the container of the simulator, I clearly can see that a lot of swift framework files are missing. If I create an example project with Xcode, it works and also the swift framework files are present.

**Comment by ikesyo** [comment]
I tested `swift2.3` branch of Eureka with Xcode 8 beta 3, but I can't build the project within Xcode, without using Carthage. So looks like this is not related to Carthage itself, right?

**Comment by benjohnde** [comment]
It seems to be an issue of Xcode. If you copy all the swift frameworks files into your project and then create a new build phase "Copy files with Destination 'Frameworks'" containing all those siwft framework files, you will be able to run your project properly.

The files are located here: ``/Applications/Xcode-beta.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/swift/iphonesimulator``.

``XcodeDefault.xctoolchain`` can be replaced with ``Swift_2.3.xctoolchain`` if you want to build/link against Swift 2.3 instead of Swift 3.0 and ``iphonesimulator`` with ``iphoneos`` if that's relevant to you.

I just created a folder/group called ``tmp-swift-libraries`` for later removal and also added the folder to the gitignore. One may create a small run script for copying those libraries before the actual build if not existing.

<img width="666" alt="screen shot 2016-07-20 at 15 56 57" src="https://cloud.githubusercontent.com/assets/5439810/16989111/317675d2-4e93-11e6-8890-3b6312113fb4.png">

It is a pretty dirty workaround. Guess this bug will be fixed soon as some posts are also existing in the apple dev forums.

**Comment by guidomb** [comment]
I don't know if this is related but I am having a problem trying to build CocoaLumberjack with Carthage and XCode 8. I've submitted https://github.com/CocoaLumberjack/CocoaLumberjack/issues/755 but I think this might be a Carthage issue. CocoaLumberjack builds correctly if I use xcodebuild directly.

Using the XCode 7.3.1 toolchain I get the following output:

```
➜  CocoaLumberjack git:(ed86d69) xcodebuild -version
Xcode 7.3.1
Build version 7D1014
➜  CocoaLumberjack git:(ed86d69) carthage build --no-skip-current --platform ios
*** xcodebuild output can be found in /var/folders/gf/dslbq4t94zg44sz7wnk2lg2r0000gp/T/carthage-xcodebuild.GoJy54.log
*** Building scheme "CocoaLumberjack" in Lumberjack.xcworkspace
*** Building scheme "CocoaLumberjackSwift" in Lumberjack.xcworkspace
*** Building scheme "CocoaLumberjackSwift-iOS" in Lumberjack.xcworkspace
*** Building scheme "CocoaLumberjack-iOS" in Lumberjack.xcworkspace
``` 


Using the XCode 8.0 toolchain (Swift 2.3) I get the following output:

```
➜  CocoaLumberjack git:(ed86d69) xcodebuild -version                            
Xcode 8.0
Build version 8S174q
➜  CocoaLumberjack git:(ed86d69) TOOLCHAINS=com.apple.dt.toolchain.Swift_2_3 carthage build --no-skip-current --platform ios
*** xcodebuild output can be found in /var/folders/gf/dslbq4t94zg44sz7wnk2lg2r0000gp/T/carthage-xcodebuild.YilbB2.log
xcodebuild timed out while trying to read CoreDataLogger.xcodeproj 
```

Also the xcodebuild output file was empty.

Judging by the console output my guest is that for some reason Carthage is picking up some workspace file which it shouldn't use because the project's root directory only contains a `Lumberjack.xcodeproj`. For some reason this use to work with XCode 7.3.1 but not with XCode 8.0 because `CoreDataLogger.xcodeproj ` seems to be a Demo project and shouldn't be built when building the framework.

**Comment by mdiep** [comment]
> ```
> xcodebuild timed out while trying to read CoreDataLogger.xcodeproj
> ```

That's an Xcode issue that we can't really work around. 😕

**Comment by guidomb** [comment]
Are you sure is an Xcode issue. I mean obviously is `xcodebuild` the one that timed out but if I do

```
TOOLCHAINS=com.apple.dt.toolchain.Swift_2_3 xcodebuild -project Lumberjack.xcodeproj -scheme CocoaLumberjack-iOS build && TOOLCHAINS=com.apple.dt.toolchain.Swift_2_3 xcodebuild -project Lumberjack.xcodeproj -scheme CocoaLumberjackSwift-iOS build
```

The project compiles. What is carthage build doing apart from what the command above does?

**Comment by ikesyo** [comment]
`xcodebuild -project Demos/CoreDataLogger/CoreDataLogger.xcodeproj -showBuildSettings` success with Xcode 7.3.1 but fails with Xcode 8. So this would be an Xcode issue. Closing.

> What is carthage build doing apart from what the command above does?

Carthage collects all the shared frameworks schemes of all the `*.xcodeproj` or `*.xcworkspace`.

**Comment by guidomb** [comment]
@ikesyo Thanks for the explanation! I will file a radar and hope this gets fixed in the new beta. In the mean (probably for different thread), shouldn't carthage provide a way to specify the project/workspace from which to collect the shared frameworks schemes?

**Comment by salling** [comment]
Re: "xcodebuild -showBuildSettings"

I have some additional info about what seems to trigger this (but no workaround):
https://forums.developer.apple.com/message/153599

**Comment by jtomchak** [comment]
I'm having the same issue with Xcode 8

```
xcodebuild timed out while trying to read CoreDataLogger.xcodeproj �
________________________________________________________________________________
| => xcodebuild -version
Xcode 8.0
Build version 8A218a
____________________
```
no workaround yet. :-(

**Comment by mdiep** [comment]
That issue will hopefully be fixed in the next Carthage release sometime this weekend.

**Comment by jtomchak** [comment]
Thanks for the update. Let me know if I can help?

**Comment by rgmvisser** [comment]
For me this worked: `xcodebuild clean -showBuildSettings`

**Comment by jtomchak** [comment]
I'll give that a try today. I thought I ran that once, but I tried so many things who knows. I'll post the results.

**Comment by ikesyo** [comment]
#1447 was merged and releases as 0.18. So the hang should not happen anymore.

**Comment by jtomchak** [comment]
I can confirm that 0.18 fixed the hanging error for me. Thanks guys for the hard work!
